crates/store/re_data_loader/src/mcap/schema/sensor_msgs/point_cloud_2.rs

```diff
@@ -0,0 +1,394 @@
+use std::io::Cursor;
+
+use arrow::{
+    array::{
+        ArrayBuilder, BooleanBuilder, FixedSizeListBuilder, ListBuilder, StringBuilder,
+        StructBuilder, UInt8Builder, UInt32Builder,
+    },
+    datatypes::{DataType, Field, Fields},
+};
+use byteorder::{BigEndian, LittleEndian, ReadBytesExt as _};
+use re_chunk::{Chunk, ChunkComponents, ChunkId, TimePoint};
+use re_log_types::TimeCell;
+use re_mcap_ros2::sensor_msgs::{self, PointField, PointFieldDatatype};
+use re_types::{
+    AsComponents as _, Component as _, ComponentDescriptor, SerializedComponentColumn, archetypes,
+    components,
+};
+use std::collections::HashMap;
+
+use crate::mcap::{
+    cdr,
+    decode::{McapMessageParser, ParserContext, PluginError, SchemaPlugin},
+};
+
+/// Plugin that parses `sensor_msgs/msg/PointCloud2` messages.
+#[derive(Default)]
+pub struct PointCloud2SchemaPlugin;
+
+impl SchemaPlugin for PointCloud2SchemaPlugin {
+    fn name(&self) -> crate::mcap::decode::SchemaName {
+        "sensor_msgs/msg/PointCloud2".into()
+    }
+
+    fn parse_schema(
+        &self,
+        _channel: &mcap::Channel<'_>,
+    ) -> Result<re_sorbet::SorbetSchema, PluginError> {
+        Err(PluginError::Other(anyhow::anyhow!("todo")))
+    }
+
+    fn create_message_parser(
+        &self,
+        _channel: &mcap::Channel<'_>,
+        num_rows: usize,
+    ) -> Box<dyn crate::mcap::decode::McapMessageParser> {
+        Box::new(PointCloud2MessageParser::new(num_rows)) as Box<dyn McapMessageParser>
+    }
+}
+
+fn fixed_size_list_builder<T: ArrayBuilder + Default>(
```

Comment:
nit: Might be time to move this to a separate util file, since its used in multiple files now